cmd/gen.go

```diff
@@ -36,67 +37,89 @@ var genCmd = &cobra.Command{
 	# run it
 	iter8 run
 
-	# generate formatted output
+	# generate formatted text output
 	iter8 gen
 `,
 	Run: func(cmd *cobra.Command, args []string) {
-		var tmpl *template.Template
-		var err error
-
-		switch strings.ToLower(outputFormat) {
-		case "text":
-			tmpl, err = builtInTemplates["text"].Parse("{{ formatText . }}")
-			if err != nil {
-				log.Logger.WithStackTrace(err.Error()).Error("unable to parse text template")
-				os.Exit(1)
-			}
-
-		case "custom":
-			// read in the template file
-			tplBytes, err := ioutil.ReadFile(templateFilePath)
-			if err != nil {
-				log.Logger.WithStackTrace(err.Error()).Error("unable to read template file")
-				os.Exit(1)
-			}
-
-			// ensure it is a valid template
-			tmpl, err = template.New("tpl").Funcs(template.FuncMap{
-				"toYAML": toYAML,
-			}).Option("missingkey=error").Funcs(sprig.TxtFuncMap()).Parse(string(tplBytes))
-			if err != nil {
-				log.Logger.WithStackTrace(err.Error()).Error("unable to parse template file")
-				os.Exit(1)
-			}
-
-		default:
-			log.Logger.Error("invalid output format; valid formats are: text | custom")
-			os.Exit(1)
-		}
-
-		// build experiment
 		log.Logger.Trace("build started")
-		exp, err := build(true)
+		// build experiment
+		// replace FileExpIO with ClusterExpIO to build from cluster
+		fio := &FileExpIO{}
+		exp, err := Build(true, fio)
 		log.Logger.Trace("build finished")
 		if err != nil {
 			log.Logger.Error("experiment build failed")
 			os.Exit(1)
 		}
 
-		// execute template
-		var b bytes.Buffer
-		err = tmpl.Execute(&b, exp)
+		// generate formatted output
+		err = exp.Gen(outputFormat)
 		if err != nil {
-			log.Logger.WithStackTrace(err.Error()).Error("unable to execute template")
 			os.Exit(1)
 		}
-
-		// print output
-		fmt.Println(b.String())
 	},
 }
 
+// Gen creates output from experiment as per outputFormat
+func (exp *Experiment) Gen(outputFormat string) error {
+	var tmpl *template.Template
+	var err error
+
+	switch strings.ToLower(outputFormat) {
+	case "text":
+		tmpl = builtInTemplates["text"]
+
+	case "custom":
+		// read in the template file
+		tplBytes, err := ioutil.ReadFile(templateFilePath)
+		if err != nil {
+			log.Logger.WithStackTrace(err.Error()).Error("unable to read template file")
+			return err
+		}
+
+		// ensure it is a valid template
+		tmpl, err = template.New("tpl").Funcs(template.FuncMap{
+			"toYAML": toYAML,
+		}).Option("missingkey=error").Funcs(sprig.TxtFuncMap()).Parse(string(tplBytes))
+		if err != nil {
+			log.Logger.WithStackTrace(err.Error()).Error("unable to parse template file")
+			return err
+		}
+
+	default:
```

Comment:
```go
tmpl = builtInTemplates["text"]
```

This is how registered  items are used. 

Other than reusable constants, the if statement you are suggesting is equivalent to the switch statement above. In any case, since you prefer if, I made it if.

Reply:
I am not concerned about the switch statement. I could only see that "text" was used via an explicit case in the switch. If we dynamically register a template (for example, the k8s template), the switch will not have the key. Instead we need to just try to get it and use or fail.  I wrote it as an if statement since I had just 2 cases, custom and everything else.  Dynamic registration means text is no longer special per se.